apps/frontend/src/components/technique/AssignInstrumentsToTechniques.tsx

```diff
@@ -0,0 +1,128 @@
+import Alert from '@mui/material/Alert';
+import Button from '@mui/material/Button';
+import Container from '@mui/material/Container';
+import Grid from '@mui/material/Grid';
+import Typography from '@mui/material/Typography';
+import { Form, Formik } from 'formik';
+import React from 'react';
+
+import FormikUIAutocomplete from 'components/common/FormikUIAutocomplete';
+import { InstrumentFragment } from 'generated/sdk';
+import { useInstruments } from 'hooks/instrument/useInstruments';
+import { getUniqueArray } from 'utils/helperFunctions';
+
+type AssignInstrumentsToTechniquesProps = {
+  close: () => void;
+  assignInstrumentsToTechniques: (
+    instrument: InstrumentFragment[]
+  ) => Promise<void>;
+  removeIntrumentsFromTechnique: (id: number[]) => Promise<void>;
+  instrumentIds: (number | null)[];
+};
+
+const AssignInstrumentsToTechniques = ({
+  close,
+  assignInstrumentsToTechniques,
+  removeIntrumentsFromTechnique,
+  instrumentIds,
+}: AssignInstrumentsToTechniquesProps) => {
+  const { instruments, loadingInstruments } = useInstruments();
+
+  const uniqueInstrumentIds = getUniqueArray(instrumentIds);
+
+  return (
+    <Container
+      component="main"
+      maxWidth="xs"
+      data-cy="technique-instruments-assignment"
+    >
+      <Formik
+        initialValues={{
+          selectedInstrumentIds: uniqueInstrumentIds || null,
+        }}
+        onSubmit={async (values): Promise<void> => {
+          const selectedInstruments = instruments.filter((instrument) =>
+            values.selectedInstrumentIds.find(
+              (selectedInstrumentId) => selectedInstrumentId === instrument.id
+            )
+          );
+
+          if (instrumentIds.length != 0) {
+            const addedInstruments = selectedInstruments.filter(
+              (instrument) =>
+                !instrumentIds.find(
+                  (selectedInstrumentId) =>
+                    selectedInstrumentId === instrument.id
+                )
+            );
+            const deletedInstruments: number[] = [];
+
+            instrumentIds.forEach((id) => {
+              if (
+                id != null &&
+                !selectedInstruments.find(
+                  (selectedInstrumentId) => selectedInstrumentId.id === id
+                )
+              ) {
+                deletedInstruments.push(id);
+              }
+            });
+
+            await assignInstrumentsToTechniques(addedInstruments || null);
+
+            await removeIntrumentsFromTechnique(deletedInstruments);
```

Comment:
Should we do a IF to check if `deletedInstruments` is `[]` or not, before calling `removeIntrumentsFromTechnique`.